doc/40-Configuration-and-Resources.md

```diff
@@ -0,0 +1,267 @@
+
+To begin we create a simple Form at `application/forms/DatabaseConfigForm.php`. We will use this Form to set the name of our currently hard-coded `assets.sqlite` database file.
+
+```php
+<?php
+
+namespace Icinga\Module\Training\Forms;
+
+use ipl\Web\Compat\CompatForm;
+
+class DatabaseConfigForm extends CompatForm
+{
+    protected function assemble()
+    {
+        $this->addElement('text', 'dbfile', ['label' => 'Database File']);
+
+        $this->addElement(
+            'submit',
+            'submit',
+            [
+                'label' => $this->translate('Save Changes')
+            ]
+        );
+    }
+}
+```
+
+This will store the given text value in a `dbfile` key (`dbfile = foobar`).
+
+The next step is a `ConfigController` with a `databaseAction()` that will provide the endpoint `config/database`.
+
+Within the Controller we use the `Icinga\Application\Config` class to read and write the config data.
+This class has a static function that can be used to load a module's configuration `Config::module()`.
+
+```php
+$moduleConfig = Config::module('training');
+$moduleConfig->getSection('database')
+```
+
+We can get/set an INI section and then store it to the module's configuration file.
+
+```php
+$moduleConfig->setSection('database', $form->getValues());
+$moduleConfig->saveIni();
+```
+
+The ConfigController could look like this:
+
+```php
+<?php
+
+namespace Icinga\Module\Training\Controllers;
+
+use Icinga\Module\Training\Forms\DatabaseConfigForm;
+
+use Icinga\Application\Config;
+use Icinga\Web\Widget\Tab;
+use Icinga\Web\Widget\Tabs;
+use Icinga\Web\Notification;
+use ipl\Web\Compat\CompatController;
+
+class ConfigController extends CompatController
+{
+    public function init()
+    {
+        $this->assertPermission('config/modules');
+        parent::init();
+    }
+
+    public function databaseAction()
+    {
+        $moduleConfig = Config::module('training');
+        $form = new DatabaseConfigForm();
+        $form->populate($moduleConfig->getSection('database'))
+            ->on(DatabaseConfigForm::ON_SUCCESS, function ($form) use ($moduleConfig) {
+                $moduleConfig->setSection('database', $form->getValues());
+                $moduleConfig->saveIni();
+
+                Notification::success('New configuration has successfully been stored');
+            })->handleRequest($this->getServerRequest());
+
+        $this->mergeTabs($this->Module()->getConfigTabs()->activate('database'));
+
+        $this->addContent($form);
+    }
+
+    /**
+     * So that we see the Module Tab and the config tab together
+     */
+    protected function mergeTabs(Tabs $tabs): void
+    {
+        foreach ($tabs->getTabs() as $tab) {
+            $this->tabs->add($tab->getName(), $tab);
+        }
+    }
+}
+```
+
+This will load the current values from from the `database` section and populate the DataBaseConfigForm with it. On saving it will store the given values in the `database` section.
```

Comment:
```suggestion
This will load the current values from the `database` section and populate the DataBaseConfigForm with it. On saving, it will store the given values in the `database` section.
```